Zend/zend_closures.c

```diff
@@ -361,11 +361,12 @@ static zend_result zend_create_closure_from_callable(zval *return_value, zval *c
 
 		memset(&call, 0, sizeof(zend_internal_function));
 		call.type = ZEND_INTERNAL_FUNCTION;
-		call.fn_flags = mptr->common.fn_flags & ZEND_ACC_STATIC;
+		call.fn_flags = mptr->common.fn_flags & (ZEND_ACC_STATIC | ZEND_ACC_DEPRECATED);
 		call.handler = zend_closure_call_magic;
 		call.function_name = mptr->common.function_name;
 		call.scope = mptr->common.scope;
 		call.doc_comment = NULL;
+		call.attributes = mptr->common.attributes;
```

Comment:
I don't quite understand, why I do not need to increase the refcount here: If I do, I get a leak.

Reply:
`call` isn't a trampoline, and the function `call` is freed in `zend_closure_free_storage`. The attributes are ignored by `zend_closure_free_storage`. So you're counting on that the attributes of the original function outlive the closure instance.

Reply:
Now I actually wonder why we refcount the attributes for trampolines in the first place. They always come from a real function, and real functions should outlive trampolines I think. Maybe I'm missing something...